crates/biome_js_analyze/tests/specs/nursery/useExportType/invalid.ts.snap

```diff
@@ -0,0 +1,180 @@
+
+export {}
+
+```
+
+# Diagnostics
+```
+invalid.ts:2:10 lint/nursery/useExportType  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! Several exports are only types and should thus use export type.
+  
+    1 │ import { type T1, V1 } from "./mod.ts";
+  > 2 │ export { T1, V1 };
+      │          ^^
+    3 │ 
+    4 │ import type { T2, T3 } from "./mod.ts";
+  
+  i Using export type allows transpilers to safely drop exports of types without looking for their definition.
+  
+  i Safe fix: Use inline type exports.
+  
+    2 │ export·{·type·T1,·V1·};
+      │          +++++         
+
+```
+
+```
+invalid.ts:5:8 lint/nursery/useExportType  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! All exports are only types and should thus use export type.
+  
+    4 │ import type { T2, T3 } from "./mod.ts";
+  > 5 │ export { T2, T3 };
+      │        ^^^^^^^^^^^
+    6 │ 
+    7 │ import type T4 from "./mod.ts";
+  
+  i Using export type allows transpilers to safely drop exports of types without looking for their definition.
+  
+  i Safe fix: Use a grouped export type.
+  
+    5 │ export·type·{·T2,·T3·};
+      │        +++++           
+
+```
+
+```
+invalid.ts:8:8 lint/nursery/useExportType  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! All exports are only types and should thus use export type.
+  
+     7 │ import type T4 from "./mod.ts";
+   > 8 │ export { T4 };
+       │        ^^^^^^^
+     9 │ 
+    10 │ import type * as ns from "./mod.ts";
+  
+  i Using export type allows transpilers to safely drop exports of types without looking for their definition.
+  
+  i Safe fix: Use a grouped export type.
+  
+    8 │ export·type·{·T4·};
+      │        +++++       
+
+```
+
+```
+invalid.ts:11:8 lint/nursery/useExportType  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! All exports are only types and should thus use export type.
+  
+    10 │ import type * as ns from "./mod.ts";
+  > 11 │ export { ns };
+       │        ^^^^^^^
+    12 │ 
+    13 │ interface Interface {}
+  
+  i Using export type allows transpilers to safely drop exports of types without looking for their definition.
+  
+  i Safe fix: Use a grouped export type.
+  
+    11 │ export·type·{·ns·};
+       │        +++++       
+
+```
+
+```
+invalid.ts:18:10 lint/nursery/useExportType  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  ! Several exports are only types and should thus use export type.
+  
+    16 │ function func() {}
+    17 │ class Class {}
+  > 18 │ export { Interface, TypeAlias, Enum, func as f, Class };
+       │          ^^^^^^^^^^^^^^^^^^^^
+    19 │ 
+    20 │ export /*0*/ { /*1*/ type /*2*/ func /*3*/, /*4*/ type Class as C /*5*/ } /*6*/;
+  
+  i Using export type allows transpilers to safely drop exports of types without looking for their definition.
+  
+  i Safe fix: Use inline type exports.
+  
+    18 │ export·{·type·Interface,·type·TypeAlias,·Enum,·func·as·f,·Class·};
```

Comment:
Can a bundler **really** three shake this? Or can typescript **really** do that?

Reply:
I think so. Since the export is declared as a type, the bundler/compiler knows that the entity is not used a runtime. Thus, it can remove this specific export. Otherwise, it has to check if the definition is a type. Many bundlers, such as ESbuil assumes that you use the isolated module mode that enforces the usage of type-only imports and exports for types. Thus, they don't check definitions and only rely on the type-only imports/exports.